### PR TITLE
chore: show agents how to view test code coverage

### PR DESCRIPTION
**Pull request by wjones127**
Providing some scripts and instructions for how to view coverage. This is usable by coding agents, which should enable them to help us improve our test coverage.

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -135,6 +139,8 @@ Tests:
   /// # }
   /// ```
   ```
+* Code coverage can be skipped for test utilities and non-critical paths using
```
Comment (westonpace) [inline AGENTS.md:142]: Is there any way to just skip code coverage for anything in the test cfg?  Or is that already done and this is for test utilities that are in main scope for some reason?

Comment (wjones127) [reply AGENTS.md:142]: Doesn't seem like there's a particularly easy way. https://github.com/taiki-e/cargo-llvm-cov/issues/123

But most of the time tests have their lines covered. It's just some edge cases like branches that only happen on test failures that get missed. I think that's okay.

**Review by westonpace — APPROVED** [review]
